docker-compose-mongo.yml

```diff
@@ -0,0 +1,88 @@
+# NOTE: This docker-compose spins up kafka, MongoDB, and a kafka-connect pod. This does NOT start up the S3-depositor module.
+version: '3'
+services:
+  zookeeper:
+    image: wurstmeister/zookeeper
+    ports:
+      - "2181:2181"
+
+  kafka:
+    image: wurstmeister/kafka
+    ports:
+      - "9092:9092"
+    environment:
+      KAFKA_ADVERTISED_HOST_NAME: ${DOCKER_HOST_IP}
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
+      KAFKA_CREATE_TOPICS: "testing:1:1"
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+
+  mongodb_container:
+    image: mongo:6
+    environment:
+      - MONGO_REPLICA_SET_NAME=rs0
+      - MONGO_IP=${MONGO_IP}
+    ports:
+      - "27017:27017"
+    volumes:
+      - mongodb_data_container:/MongoDB_data/db
+    healthcheck:
+      test: | 
+        test $$(mongosh --quiet --eval "try { rs.initiate({ _id: 'rs0', members: [{ _id: 0, host: '${MONGO_IP}' }] }).ok } catch (_) { rs.status().ok }") -eq 1
+      interval: 10s
+      start_period: 30s
+    command: ["--replSet", "rs0", "--bind_ip_all"]
+
+
+  kafka-connect:
+    image: cp-kafka-connect:6.1.9
+    build:
+      context: ./mongo-connector
+      dockerfile: Dockerfile
+    ports:
+      - "8083:8083"
+    depends_on:
+      mongodb_container:
+        condition: service_healthy
+    environment:
+      MONGO_URI: mongodb://${MONGO_IP}:${MONGO_PORT}/?replicaSet=rs0
+      MONGO_DB_NAME: ODE
+      CONNECT_BOOTSTRAP_SERVERS: ${DOCKER_HOST_IP}:9092

```

Comment:
Is this MONGO_IP always the same as the DOCKER_HOST_IP? Maybe use one variable if yes. I got this issue testing the connector in local:
![image](https://github.com/usdot-jpo-ode/jpo-s3-deposit/assets/62157949/b366ca0b-9e31-416a-a565-8cd944f0cd0e)

![image](https://github.com/usdot-jpo-ode/jpo-s3-deposit/assets/62157949/79f2ce7b-17db-4857-b829-64a5d944b64a)

It cannot connect to Kafka message broker.

Reply:
In some cases, we do want to be able to vary the mongo_ip for more production deployments where a true replica-set mongodB deployment is desired.
I was able to reproduce the error you are seeing when I attempted to create connectors that didn't have a kafka topic already made for them. Can you try updating the KAFKA_CREATE_TOPICS variable (line 16 in docker-compose.yml) to the following:

KAFKA_CREATE_TOPICS: "topic.OdeSpatJson:1:1,topic.OdeBsmJson:1:1,topic.OdeTimJson:1:1,topic.OdePsmJson:1:1,topic.OdeRawEncodedPsmJson,topic.OdeRawEncodedBSMJson:1:1,topic.OdeRawEncodedSPATJson:1:1,topic.OdeRawEncodedTIMJson:1:1,topic.OdeRawEncodedMAPJson:1:1,topic.OdeMapJson:1:1,topic.OdeRawEncodedPSMJson:1:1,topic.OdePsmJson:1:1"

Reply:
I have added this fix to a current PR in the CDOT fork.

Reply:
Updated the create topics list. Still in the loop waiting for 200. Also tried curl command in host machine, I got this result:
```
dev@dev:~$ curl http://connect:8083/connectors
curl: (6) Could not resolve host: connect

```

Reply:
Oh I see, localhost will not work. This will cause the kafka connect image to try to look at its own 9092 port for the kafka broker instead of through your host's network. Try runing 'ifconfig' in a wsl terminal and use the docker0 network or eth0 network IP.

Reply:
I tried the docker0 IP, and it connected to Kafka message broker. However, the PUT request from the .sh script return 400 response within duration of 2 mins after containers are launched. 
![image](https://github.com/usdot-jpo-ode/jpo-s3-deposit/assets/62157949/fc0b6b08-1540-44c8-921c-05386ddcde33)

Reply:
Interesting, can you share your current .env file and run the following commands for me in a WSL terminal and let me know the results:
sudo apt-get update
sudo apt-get install kafkacat
kafkacat -b localhost:9092 -G test_consumer_group topic.OdeSpatJson -f '\nTopic: %t\nKey: %k\nTimestamp: %T\nValue: %s\n' -o beginning

install mongosh in wsl/windows: https://www.mongodb.com/docs/mongodb-shell/install/
mongosh "mongodb://localhost:27017/?DirectConnection=true"

Also, the changes mentioned previously have now been merged into this PR so if you pull, your .env must be updated with the sample.env changes.

Reply:
I am using ubuntu 22.04 Jammy distribution not windows. 
Pulled the latest dev branch and ran the command to build images and launch containers:  
` docker-compose -f docker-compose-mongo.yml up`
It seems up, then bring it down and up again. I got below issues:
![image](https://github.com/usdot-jpo-ode/jpo-s3-deposit/assets/62157949/9124d08c-6777-4df1-8304-bc7c00f0f3a5)

Reply:
![image](https://github.com/usdot-jpo-ode/jpo-s3-deposit/assets/62157949/855a21c6-f076-404a-b94f-571f2532f990)

Reply:
@dan-du-car We have just pushed some updates that should fix that exit code and other related issues.

Reply:
Look good. Thank you!